DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_blocked_port_stp_rstp.py

```diff
@@ -0,0 +1,164 @@
+async def test_lacp_root_port(testbed, version):
+    """
+    Test Name: LACP STP/RSTP interaction
+    Test Suite: suite_functional_lacp
+    Test Overview: Test root port selection based on port priority
+    Test Procedure:
+    1. Create 2 bridge entities 4 bonds and set link up on them
+    2. Enslave ports to bonds, bonds to bridges
+    3. Change the MAC addresses for all bridges.
+    4. Set link up on all participant ports, bonds, bridges
+    5. Wait until topology converges
+    6. Verify the bridge root is bridge_1 by the lowest MAC rule
+    Verify other bridges do not consider themselves as root-bridge
+    7. Verify bridge_2 has a blocking port
+    8. Verify bridge_1 doesn't have any blocking port.
+    9. Change the priority of the port in bridge_1 which is connected to the blocking bond in bridge_2 to
+    a priority value less than the default set
+    10. Wait for topology to re-build.
+    11. Verify bridge_2 has a blocking port and is not the same port as before
+    """
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        pytest.skip(
+            'The testbed does not have enough dent with tgen connections')
+    device = dent_devices[0]
+
+    dent = device.host_name
+    bonds = {}
+    for idx, port in enumerate(device.links_dict[dent][0][:2] + device.links_dict[dent][1][:2]):
+        bonds[f'bond_{idx+1}'] = port
+    bond_to_bridges = list(bonds.keys())
+    bridges = {
+        'bridge_1': bond_to_bridges[:2],
+        'bridge_2': bond_to_bridges[2:]
+    }
+    bridge_names = list(bridges.keys())
+    hw_mac = ['22:AA:98:EA:D2:43', '44:0A:D1:68:4E:B9']
+    time_to_wait = 40 if version == 'stp' else 20
+
+    # 1. Create 2 bridge entities4  bonds and set link up on them
+    out = await IpLink.add(input_data=[{dent: [{'device': bond, 'type': 'bond', 'mode': '802.3ad'} for bond in bonds]}])
+    assert out[0][dent]['rc'] == 0, 'Failed to add bond'
+
+    out = await IpLink.add(input_data=[{dent: [{
+        'device': bridge,
+        'type': 'bridge',
+        'stp_state': 1} for bridge in bridges]}])
+    assert out[0][dent]['rc'] == 0, 'Failed to add bridge'
+
+    out = await Mstpctl.add(input_data=[{dent: [{'bridge': bridge} for bridge in bridges]}])
+    assert out[0][dent]['rc'] == 0, 'Failed to add bridge with mstpd'
+
+    out = await Mstpctl.set(input_data=[{dent: [{
+        'bridge': bridge,
+        'parameter': 'forcevers',
+        'version': version} for bridge in bridges]}])
+    assert out[0][dent]['rc'] == 0, 'Failed to set stp/rstp version'
+
+    # 2. Enslave ports to bonds, bonds to bridges
+    out = await IpLink.set(input_data=[{dent: [{'device': port, 'operstate': 'down'} for port in bonds.values()]}])
+    assert out[0][dent]['rc'] == 0, 'Failed setting links to state down'
+
+    out = await IpLink.set(input_data=[{dent: [{'device': bond, 'master': port}]} for port, bond in bonds.items()])
```

Comment:
`bonds` is a dict of 'bond name' = 'slave name':
```python
bonds[f'bond_{idx+1}'] = port
bonds.items()  # ((f'bond_{idx+1}', port), ...)
```
but you are using the wrong naming here:
```suggestion
    out = await IpLink.set(input_data=[{dent: [{'device': port, 'master': bond}]} for bond, port in bonds.items()])
```